app/experimenter/nimbus-ui/src/lib/experiment.ts

```diff
@@ -5,37 +5,32 @@
 import { RedirectCheck } from "../components/AppLayoutWithExperiment";
 import { getAllExperiments_experiments } from "../types/getAllExperiments";
 import { getExperiment_experimentBySlug } from "../types/getExperiment";
-import { NimbusExperimentStatus } from "../types/globalTypes";
+import {
+  NimbusExperimentPublishStatus,
+  NimbusExperimentStatus,
+} from "../types/globalTypes";
 
 export function getStatus(
   experiment?: getExperiment_experimentBySlug | getAllExperiments_experiments,
 ) {
   const status = experiment?.status;
-
-  // The experiment is in review or accepted state
-  const preparation = [
-    NimbusExperimentStatus.REVIEW,
-    NimbusExperimentStatus.ACCEPTED,
-  ].includes(status!);
+  const publishStatus = experiment?.publishStatus;
 
   // The experiment is or was out in the wild (live or complete)
   const released = [
     NimbusExperimentStatus.LIVE,
     NimbusExperimentStatus.COMPLETE,
   ].includes(status!);
 
-  // The experiment's fields generally cannot be updated (accepted, live, or complete)
-  const locked = released || NimbusExperimentStatus.ACCEPTED === status!;
-
   return {
     draft: status === NimbusExperimentStatus.DRAFT,
     preview: status === NimbusExperimentStatus.PREVIEW,
-    review: status === NimbusExperimentStatus.REVIEW,
-    accepted: status === NimbusExperimentStatus.ACCEPTED,
     live: status === NimbusExperimentStatus.LIVE,
     complete: status === NimbusExperimentStatus.COMPLETE,
-    locked,
-    preparation,
+    idle: publishStatus === NimbusExperimentPublishStatus.IDLE,
+    approved: publishStatus === NimbusExperimentPublishStatus.APPROVED,
+    review: publishStatus === NimbusExperimentPublishStatus.REVIEW,
+    waiting: publishStatus === NimbusExperimentPublishStatus.WAITING,
     released,
```

Comment:
An experiment can be in any combination of `status` and `publishStatus` simultaneously so I'm not sure that treating them flatly like this will work?

Reply:
Similar to the `released` one, we can combine them as needed, they don't need to compare a single value. This helper is basically meant to serve as a shorthand for retrieving the (previously just lifecycle, but now lifecycle and publish) status so you don't have the drag the long `NimbusExperimentStatus` and `NimbusExperimentPublishStatus` around everywhere you go.

What I would really like is for us to rename everything called `status` to `lifecycyleStatus` or similar on the frontend and backend, because I think a function like this _is_ helpful, it's just confusing to know, when talking about "status", what exactly you mean.

Reply:
Right but if you treat them flatly how do you encode the difference between Draft/Review and Live/Review?

Reply:
Kind of a same note on https://github.com/mozilla/experimenter/pull/4946/files#r604440379, I do really like the idea of moving some of this logic to the BE but maybe in another ticket?

Reply:
Filed as https://jira.mozilla.com/browse/EXP-1158